### PR TITLE
[D1] fix `wrangler d1 migration apply` to for queries not ending in a semi-colon

### PR DESCRIPTION
**Pull request by rozenmd**
Prior to this PR, migrations such as:
```
SELECT 1
```

would fail with a nonsensical error about INSERTS: `✘ [ERROR] near "INSERT": syntax error at offset 199 [code: 7500]`

What was happening was that wrangler was injecting a statement at the end of the migration, to update the migrations table saying the query successfully executed.

To avoid this, this PR moves the statement to the start of the migration.

### CODE REVIEW

File: .changeset/gorgeous-taxis-argue.md
Comment (rozenmd) [inline .changeset/gorgeous-taxis-argue.md:5]:
```suggestion
fix: fix `wrangler d1 migrations apply` to allow migrations that dont end in a semicolon

Prior to this PR, migrations such as: `SELECT 1` would fail with a nonsensical error about INSERTS: `✘ [ERROR] near "INSERT": syntax error at offset 199 [code: 7500]`

What was happening was that wrangler was injecting a statement at the end of the migration, to update the migrations table saying the query successfully executed.

To avoid this, this PR moves the statement to the start of the migration.
```

File: .changeset/gorgeous-taxis-argue.md
Comment (justin-mp) [inline .changeset/gorgeous-taxis-argue.md:5]:
```suggestion
fix: fix `wrangler d1 migrations apply` to allow migrations that don't end in a semicolon
```

File: packages/wrangler/src/d1/migrations/apply.tsx
Comment (justin-mp) [inline packages/wrangler/src/d1/migrations/apply.tsx:149]: Why is it safe to update the migrations table to say we've completed the migration *before* the migration happens?  Are we relying on transactions somehow?

**Review by justin-mp — COMMENTED** [review]
Out of curiosity, why do we want to allow folks to have migrations with incomplete statements?  That seems buggy to me.

File: packages/wrangler/src/d1/migrations/apply.tsx
Comment (rozenmd) [reply packages/wrangler/src/d1/migrations/apply.tsx:149]: My assumption is that a query string is executed as a single query, so if one of the queries fail, the DO throws... right @geelen ?

**Comment by rozenmd** [comment]
> Out of curiosity, why do we want to allow folks to have migrations with incomplete statements? That seems buggy to me.
@justin-mp SQL statements that don't terminate in a semi-colon are still valid queries: 
image

**Comment by justin-mp** [comment]
> > Out of curiosity, why do we want to allow folks to have migrations with incomplete statements? That seems buggy to me.
> 
> @justin-mp SQL statements that don't terminate in a semi-colon are still valid queries

Interesting.  According to https://sqlite.org/syntax/sql-stmt-list.html, it looks like the semicolon is indeed optional. 

However, the `sqlite3` command seems to require it

```
$ sqlite3 
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> select 1
   ...>
```
(It won't accept the command until I pass in a `;`.)

**Comment by rozenmd** [comment]
Worth noting that the sqlite CLI and sqlite core have slightly different implementations

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Just as a retro to this change... @rozenmd - can you write down why there were no test additions for this change?

**Comment by rozenmd** [comment]
No reason @petebacondarwin - I was waiting for a reply from @geelen before merging: https://github.com/cloudflare/workers-sdk/pull/5207#discussion_r1517827599

**Comment by petebacondarwin** [comment]
Oh! So should we actually revert this merge until the questions have been resolved?

**Comment by rozenmd** [comment]
I think that'd be best @petebacondarwin - then I can try write a test for it too

File: packages/wrangler/src/d1/migrations/apply.tsx
Comment (geelen) [reply packages/wrangler/src/d1/migrations/apply.tsx:149]: Wow I guess I missed this at the time (I am cleaning up some old tabs), but yes. Anything submitted in a single query to a D1 is executed in a transaction.

**Pull request merged**
